### DISCUSSION TITLE
Getting a route out of a gravel tour track

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#273] [Q&A] [answered]
Last year I recorded a gravel tour on the motorcycle with K2. This I would now like to drive again. But not just simply following the track, as most Garmin users do. However, it is virtually impossible to create a route from this track, because parts of the paths in OSM are marked as not passable or simply do not exist. 

Now I set the vehicle in Cruiser to _Mountainbike_, imported the track, formed the route by around 180 shaping points and exported it in gpx format with navigation. Even with _Mountainbike_ some parts of the route can only be bridged with "beeline". Here the result: [1n.zip](https://github.com/devemux86/cruiser/files/12281181/1n.zip)

Do I now have to ride this route with _Mountainbike_ or can I also use _Motorcycle_ for this? This will only be relevant if I am forced to recalculate due to extraordinary locks. This result of recalculation would then be different, although only to the next waypoint, right?

Btw: I'm starting to think about planning my bike trips in Komoot but using Cruiser for navigation. First tests are planned.

**Comment by devemux86** [comment]
@WolfgangZ42 very interesting report!

Do you use BRouter (offline) in route calculation?

> This I would now like to drive again.

With motorcycle or bike?
Regular or mountain models?

> Do I now have to ride this route with Mountainbike or can I also use Motorcycle for this?

We can use any app profile to navigate a route.

> This result of recalculation would then be different, although only to the next waypoint, right?

Correctly: _navigation rerouting only calculates until the next waypoint_.

---

If the route calculation is with BRouter (offline) then it's worth checking the OSM tags
and see if the parts can be driven by some profile options that allow using these tags?

e.g. option "Use gravel roads" allows the use of `highway=track` with `tracktype=grade1`
If "Avoid unpaved roads" is not enabled then the full `highway=track` is used (all grades).

These tests can be easily done on BRouter website with the driving-cruiser profile:
- https://github.com/devemux86/cruiser/discussions/272

**Reply by WolfgangZ42** [reply]
Yes - I exclusively use BRouter (offline)
With motorcycle - of course 😎 and it's an enduro model.

**Reply by WolfgangZ42** [reply]
And yes I've enabled "Use gravel roads" and "Ignore closed roads". Sure I didn't use "Avoid unpaved roads".

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the extra details.

You mentioned bike trips in Komoot, so I thought you meant the bike profiles.

> With motorcycle - of course sunglasses and it's an enduro model.

> I've enabled "Use gravel roads" and "Ignore closed roads". Sure I didn't use "Avoid unpaved roads".

So you mean the `driving-cruiser` profile.
If the current options do not use some road / track types, we can check the tags and think for improvements.

**Comment by devemux86** [comment]
> I'm starting to think about planning my bike trips in Komoot but using Cruiser for navigation.

If you mean to drive with bike and calculate the route with bike profile,
Cruiser uses the original BRouter bike profiles: fastbike and trekking.

If there is interest, I can customize these as well (like I did with the driving-cruiser profile).
e.g. adding options to use more OSM paths.

**Reply by 0709wiwiwi** [reply]
@WolfgangZ42 

Can you pse share the original record track ?

@WolfgangZ42.
Without the original track record just had a quick test anyway.
Through your edited gpx file the roads **are already matched** to the map.
I had the same file than recalculated in RouteYou web (Doesn't take long). 
Then download the result as a gpx Locus map.
You can download this file here. 
https://www.routeyou.com/nl/route/view/13246413/fietsroute/1n

I did remove all (very many) straight turn commands in the gpx file using Notepad ++.
And then put both files into one single gpx file for easy comparison. I see one discrepancy. 
 In the gpx editor you see one track in red color and the second in blue color. 
I have not checked the correctness of the instructions. 
No other manual checks for correctness are so performed !

[1nandroute_you.gpx.txt](https://github.com/devemux86/cruiser/files/12283696/1nandroute_you.gpx.txt)

**Reply by devemux86** [reply]
@0709wiwiwi there is a navigation gpx attached in the first post.

**Reply by WolfgangZ42** [reply]
> If there is interest, I can customize these as well

Thank you, not at the moment - let's wait and see what the results of the tests will be.

**Reply by WolfgangZ42** [reply]
Thanks a lot Willy - I'll look at it tomorrow, now it's to late for difficult analysises.

**Reply by WolfgangZ42** [reply]
> You can download this file here.
https://www.routeyou.com/nl/route/view/13246413/fietsroute/1n

Thanks a lot Willy! 
You saved some of my life time - it took me approximately 2 hours to set all the waypoints manually for 1n.gpx and now with RouteYou it takes just a minute. The trick is to download the Locus Map version, as you mentioned. Btw: I had a valid result with 177 waypoints (but with some beeline parts) and RouteYou set 232 of them (w/o any beeline!). Human brain beats artificial intelligence? But you removed the "overhead" already ...

Okay - I had to setup a test account for RouteYou. I'm not quite sure if I need another tool. This was a very special gravel tour with several undefined pathes - I rarely drive such routes. Most of .gpx files are easily to convert into a route manually. Now I'll use the test account to convert the other 3 days of this adventure - and save a lot of time!

Again many thanks, Willy, for this hint!

**Reply by devemux86** [reply]
> This was a very special gravel tour with several undefined paths

@WolfgangZ42 have you identified some of these paths?
It would be interesting to see their OSM tags.

**Reply by 0709wiwiwi** [reply]
@WolfgangZ42 

The reason I asked about the original record track is this Wolfgang.

When importing a record track, RouteYou converts it using the osm shortest profile.
And RouteYou recognizes straight_line sections and adds them into the navtrack.

What I doubt, however, is this.
 
I wonder if the osm shortest profile won't route you all too quickly on bike or foot trails.
I do want like to meet you but preferably not on a bike path. And you may not want that either.

That is why I suggest it more cautiously by a Cruiser_Brouter routing suggestion next.

https://github.com/devemux86/cruiser/discussions/271#discussioncomment-6645447

Use an optimal (to design) enduro biker profile and for the still non-routable parts use:

- 1. The reroute profile (similar to RouteYou shortest osm).
- 2. And as an ultimate solution use a beeline or straight line.

Whether my suspicion is entirely correct I cannot figure out on my own.
The Enduro bikers could so share their usage experiences.

Such navtrack reviews indeed do require a lot of time....time you are not on the road.

**Reply by WolfgangZ42** [reply]
@devemux86 
> ... have you identified some of these paths? It would be interesting to see their OSM tags.

Here I've a short part of my gravel tour: [1n_short.gpx.txt](https://github.com/devemux86/cruiser/files/12306712/1n_short.gpx.txt) and a picture with 3 undefined parts:
<img width="500" src="https://github.com/devemux86/cruiser/assets/121806238/eb00a2ec-4355-4e47-9b0a-76f7f69c927e"/>

**Reply by WolfgangZ42** [reply]
@0709wiwiwi 
> I do want like to meet you but preferably not on a bike path. And you may not want that either.

In principle, that's right, Willy. But in my case I know that we can ride these trails on motorcycle, because we did it last year.

**Reply by WolfgangZ42** [reply]
Okay - now I can real navigate with Cruiser the navigation gpx. Without any problem if I don't use recalculation at all. 
And the poor Garmin Boys have just to follow the track w/o announcements.

**Reply by 0709wiwiwi** [reply]
> In principle, that's right, Willy. But in my case I know that we can ride these trails on motorcycle, because we did it last year.

What I'm concerned about with such an extremely tolerant profiles as reroute is when using it in a new traject where you don't exactly know the situation from a historic experience.
This is why I advocate an optimal (to be designed) enduro bikers profile where you ultimately then indeed switch to reroute or beeline for a while if you still get stuck.  Where at such rather short non default profile track sections a designer is attentive and well aware of the risks. 

Btw Wolfgang. 
Route You also offers specific Garmin Zumo gpx route exports.  
As a gpx route with a "virtual track" hidden in the Garmin extension.
I can't test, I  have no Zumo. But your friends maybe can give it a try ?

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the gpx.

Checking OpenStreetMap, there don't seem to exist any defined paths at all in the marked areas.

Without straight lines, how we could have a routing here?
Even [reroute.brf](https://github.com/zossebart/brouter-misc/blob/master/reroute.brf) needs existing paths, that do not exist here.

**Reply by 0709wiwiwi** [reply]
Hi Emux.

Somehow RouteYou knows this and apparently they take over the record gpx part ? 
This is of course their internal recipe and method by RouteYou which we do not know.
How to ?  I did not study this any further.  Nice weather coming > So let's bike.

**Reply by WolfgangZ42** [reply]
> Route You also offers specific Garmin Zumo gpx route exports.

Thanks for the hint, I'll give it a try!

**Comment by Ciutnik** [comment, marked as answer]
@WolfgangZ42 
> it is virtually impossible to create a route from this track, because parts of the paths in OSM are marked as not passable or simply do not exist.

I strongly encourage you to put this roads on the OSM map, using your tracks 🙂 It's quite a fun job, easy and rewarding. You can do it not only for yourself (after that, very soon Cruiser will be guiding you down this road), but for your descendants and forever 😀

**Reply by devemux86** [reply]
[BRouter routing files](https://brouter.de/brouter/segments4/) are updated daily!

**Reply by WolfgangZ42** [reply]
@Ciutnik 
I'll definitely try it, because using another tool with routeyou.com makes little sense.

**Comment by WolfgangZ42** [comment]
Now I was 2 days on the road in the gravel - the routes with routeyou.com created from a recorded tracks. I have never heard so often "Dem Straßenverlauf folgen / Follow the road" (straight)! Every little dirt road was taken into account. 
A great adventure, everyone has been in the dirt once. Corrected three errors in the OSM. And once again proved that Cruiser is superior to both TomTom and Garmin. Only the two Cruiser fanboys were allowed to ride ahead (and had to swallow less dust). 
One small thing: on both day trips, Cruiser inserted ViaPoints where there were none. Not important, not disturbing, I just noticed. I have extracted parts of the routes and attached them here: [testVIA.zip](https://github.com/devemux86/cruiser/files/12365863/testVIA.zip)

**Reply by devemux86** [reply]
> One small thing: on both day trips, Cruiser inserted ViaPoints where there were none.

@WolfgangZ42 can you explain more the "insert" process?
During import or it is an export from RouteYou?

I implemented the shaping points a few years ago in Cruiser (and derivative apps!).
Not many apps and routers know the distinction between via and shaping points.

**Reply by devemux86** [reply]
@WolfgangZ42 I now see what you mean.

The GPX were generated by [RouteYou](https://www.routeyou.com/).
They contain intermediate points that are not defined as via or shaping points.

Cruiser in this case imports them as via points. I can change the "undefined" to shaping points.

**Reply by 0709wiwiwi** [reply]
@WolfgangZ42 
> I have never heard so often "Dem Straßenverlauf folgen / Follow the road" (straight)!
 
I told the RouteYou dev about this VERY BIG annoyance, but a fast response as we are used to here with Cruiser is not too be expected.
So I' ll keep my saying intact...  "Less TTS is the best TTS".

**Reply by devemux86** [reply]
Regardless of whether routers improve the generation of turn instructions.

Cruiser can post-process the turn instructions of all calculated or imported routes
and remove "Continue" instructions that are near (or not) to other turn instructions.
We need to think more about it (and the possible distance thresholds).

Checking your examples,
the "Continue" instructions are on locations where there is also another street and a "possibility" of a turn.

**Reply by 0709wiwiwi** [reply]
@devemux86 

I had a quick look at the gpx in the zip send by Wolfgang. 
I notice the multiple Straight commands, but no Via Points between Start and End ? 

The disturbing multiple Straight command should so **be tackled by the RouteYou.** No ? 
The auto switch between routed and manual paths by import of a simple track is magic.

(_As a **one time** test I did remove all Straight commands by Notepad++ but so you also delete maybe some usefull ones_.)

**Reply by devemux86** [reply]
> I notice the multiple Straight commands, but no Via Points between Start and End ?

They declare intermediate points without defining their type: via or shaping point.
Cruiser now handles them as via points. I can change the undefined to shaping points.

**Reply by 0709wiwiwi** [reply]
Can you show me such a single example location (trkpt number) please ?  And why does Cruiser announce these ? 

Aaah...I found them....here a single one @ position 11.  (Effectively a via point by Locus convention)
Another RouteYou issue.  A cure ? When ? No idea....After reporting the Straights I did not have a reply yet

![image](https://github.com/devemux86/cruiser/assets/54167201/7729b107-e43b-425e-8022-78250bc6d7db)

**Reply by devemux86** [reply]
> Can you show me such a single example location (trkpt number) please ?

testVIA1.gpx line 102,
testVIA2.gpx line 50, 874

**Reply by 0709wiwiwi** [reply]
> I can change the undefined to shaping points.

Excellent suggestion. Immediate cure by a wonder medicine.

**Reply by 0709wiwiwi** [reply]
@devemux86 For your information.
RouteYou just like the Brouter now also automatically removes unwanted u-turns caused by carelessly placed planner points that happen to end up on a side road.  RouteYou corrects within 100m. With the Cruiser_Brouter, the correction distance is adjustable. (default 40m)

**Reply by devemux86** [reply]
> I can change the undefined to shaping points.

Available in [Cruiser 3.0.31](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
